src/core/Table/TablePaginator.tsx

```diff
@@ -0,0 +1,324 @@
+      : `${startIndex}-${endIndex} of ${totalRows}`,
+  previousPage: 'Previous page',
+  nextPage: 'Next page',
+  goToPageLabel: (page: number) => `Go to page ${page}`,
+} as const;
+
+export type TablePaginatorProps = {
+  /**
+   * Control whether focusing tabs (using arrow keys) should automatically select them.
+   * Use 'manual' if tab panel content is not preloaded.
+   * @default 'auto'
+   */
+  focusActivationMode?: 'auto' | 'manual';
+  /**
+   * Array of possible page size options. When provided then shows current rows info and page size selection.
+   */
+  rowsPerPage?: number[];
+  /**
+   * Object of functions to localize pagination labels.
+   */
+  localization?: {
+    /**
+     * Function that returns a label for the page size selector.
+     * @default (size: number) => `${size} per page`
+     */
+    rowsPerPageLabel?: (size: number) => string;
+    /**
+     * Function that returns a label for the range of rows within the current page and the length of the whole data.
+     * @default
+     *  (
+     *  startIndex: number,
+     *  endIndex: number,
+     *  totalRows: number,
+     *  isLoading: boolean,
+     *  ) =>
+     *    isLoading
+     *      ? `${startIndex}-${endIndex}…`
+     *      : `${startIndex}-${endIndex} of ${totalRows}`;
+     */
+    rangeLabel?: (
+      startIndex: number,
+      endIndex: number,
+      totalRows: number,
+      isLoading: boolean,
+    ) => string;
+    /**
+     * A label for previous page button. Used for accessibility attribute.
+     * @default 'Previous page'
+     */
+    previousPage?: string;
+    /**
+     * A label for next page button. Used for accessibility attribute.
+     * @default 'Next page'
+     */
+    nextPage?: string;
+    /**
+     * Function that returns a label for page selector buttons. Used for accessibility attribute.
+     * @default (page: number) => `Go to page ${page}`
+     */
+    goToPageLabel?: (page: number) => string;
+  };
+} & TablePaginatorRendererProps &
+  Omit<CommonProps, 'title'>;
+
+/**
+ * Table paginator component. Recommended to pass to the `Table` as `paginatorRenderer` prop.
+ * @example
+ * <Table
+ *   // ...
+ *   paginatorRenderer={(props) => <TablePaginator {...props} />}
+ * />
+ */
+export const TablePaginator = (props: TablePaginatorProps) => {
+  const {
+    currentPage,
+    totalRowsCount,
+    pageSize,
+    onPageChange,
+    focusActivationMode = 'auto',
+    isLoading = false,
+    density = 'default',
+    rowsPerPage,
+    onPageSizeChange,
+    localization: userLocalization,
+    className,
+    ...rest
+  } = props;
+
+  useTheme();
+
+  const localization = React.useMemo(
+    () => ({ ...defaultLocalization, ...userLocalization }),
+    [userLocalization],
+  );
+
+  const pageListRef = React.useRef<HTMLDivElement | null>(null);
+
+  const [focusedIndex, setFocusedIndex] = React.useState<number>(currentPage);
+  React.useEffect(() => {
+    setFocusedIndex(currentPage);
```

Comment:
Is this only needed for keeping track internally? Because it doesn't actually focus the new `currentPage` automatically when it changes.

Reply:
If it would, then page would scroll down to the paginator. As I am not saving the current page inside, I thought that when navigating with keyboard in auto mode, it shouldn't focus next page if the user did not changed the prop, otherwise it will be inconsistent state between focused index and current index.

Reply:
Ok, it is not working well with `auto`...